src/open_inwoner/haalcentraal/tests/test_client.py

```diff
@@ -0,0 +1,44 @@
+import json
+
+from django.test import TestCase
+
+import requests_mock
+
+from ..api import BRP_2_1
+from .mixins import HaalCentraalMixin
+
+
+@requests_mock.Mocker()
+class BRPClientTest(HaalCentraalMixin, TestCase):
+    def test_request_content_type(self, m):
+        self._setUpMocks_v_2(m)
+        self._setUpService()
+
+        api_client = BRP_2_1()
+        response = api_client.make_request("999993847")
+
+        self.assertEqual(
+            m.request_history[0].headers["Content-Type"], "application/json"
+        )
+
+        data = {
+            "fields": [
+                "naam.geslachtsnaam",
+                "naam.voorletters",
+                "naam.voornamen",
+                "naam.voorvoegsel",
+                "geslacht.omschrijving",
+                "geboorte.plaats.omschrijving",
+                "geboorte.datum.datum",
+                "verblijfplaats.verblijfadres.officieleStraatnaam",
+                "verblijfplaats.verblijfadres.huisnummer",
+                "verblijfplaats.verblijfadres.huisletter",
+                "verblijfplaats.verblijfadres.huisnummertoevoeging",
+                "verblijfplaats.verblijfadres.postcode",
+                "verblijfplaats.verblijfadres.woonplaats",
+            ],
+            "type": "RaadpleegMetBurgerservicenummer",
+            "burgerservicenummer": ["999993847"],
+        }
+
+        self.assertEqual(response.request.body, json.dumps(data).encode("utf-8"))
```

Comment:
While this works for now I'd like to note I would've gone the other way: JSON parse the request body and compare it to the expected dict. 

JSON serialization is technically not 100% stable (there are multiple valid JSON strings possible for the same data, because of dict ordering and pretty pretting), and in the assertion you'd get a structural diff instead of a text diff.

Reply:
Good point!